### PR TITLE
Remove excessive docstring locations

### PR DESCRIPTION
**Pull request by vshampor**
### Changes
Removed a couple of recommended docstring locations from class documentation style guide.

### Reason for changes
These particular documentation locations seem to be superfluous. 

### Related tickets
N/A

### Tests
N/A

### CODE REVIEW

File: docs/styleguide/PyGuide.md [hunk 1]
```diff
@@ -582,7 +582,8 @@ def load_state(model: torch.nn.Module, state_dict_to_load: dict, is_resume: bool
 
 Classes should have a docstring below the class definition describing the class. If your class
 has public attributes, they should be documented here follow the same formatting as a function's
-params section.
+params section. The `__init__` function and other magic methods may be left without a textual description,
+if there is nothing special about this exact implementation of the magic method.
```
Comment (alexsu52) [inline docs/styleguide/PyGuide.md:586]: Could you specify what does `nothing special` mean? Maybe some samples when you should and should not add a docstring will clarify it.

**Comment by aiarkinx** [comment]
run tensorflow pre-commit tests

File: docs/styleguide/PyGuide.md (hunk 1 above)
Comment (vshampor) [reply docs/styleguide/PyGuide.md]: Added a few lines to clarify.

**Review by alexsu52 — APPROVED** [review]

File: docs/styleguide/PyGuide.md [hunk 2]
```diff
@@ -582,7 +582,7 @@ def load_state(model: torch.nn.Module, state_dict_to_load: dict, is_resume: bool
 
 Classes should have a docstring below the class definition describing the class. If your class
 has public attributes, they should be documented here follow the same formatting as a function's
-params section.
+params section.
```
Comment (andrey-churkin) [inline docs/styleguide/PyGuide.md:585]: The space at the end of the string is not necessary.

File: docs/styleguide/PyGuide.md [hunk 3]
```diff
@@ -613,6 +611,54 @@ class ModelTransformer:
...
+        self.param2 = param2
+```
+while this `__init__` requires a description of external dependencies and potential side effects of creating objects of the class:
+```python
+class ComplexKlass(BaseClass):
+    # ...
+   def __init__(self, param1: ParamType, param2: AnotherParamType):
+        """
+        *Add a brief explanation of what happens during this particular __init__, such as :*
+        The construction of this objects is dependent on the value of GLOBAL_VARIABLE...
```
Comment (andrey-churkin) [inline docs/styleguide/PyGuide.md:638]:
```suggestion
        The construction of this object is dependent on the value of GLOBAL_VARIABLE...
```

Comment (vshampor) [reply docs/styleguide/PyGuide.md:638]: Done

**Review by andrey-churkin — APPROVED** [review]

**Pull request merged**